src/web_service/cookie.cpp

```diff
@@ -0,0 +1,1051 @@
+// SPDX-FileCopyrightText: 2023 Yuzu Emulator Project
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+#include <sstream>
+
+#include "common/string_util.h"
+#include "cookie.h"
+
+namespace WebService {
+
+#define TM_YEAR_BASE 1900
+
+Cookie::Cookie(std::string& name, std::string& value) : m_name{name}, m_value{value} {}
+
+bool Cookie::isSecure() const {
+    return m_secure;
+}
+
+void Cookie::setSecure(bool enable) {
+    m_secure = enable;
+}
+
+bool Cookie::isHttpOnly() const {
+    return m_httpOnly;
+}
+
+void Cookie::setHttpOnly(bool enable) {
+    m_httpOnly = enable;
+}
+
+Cookie::SameSite Cookie::sameSitePolicy() const {
+    return m_sameSite;
+}
+
+void Cookie::setSameSitePolicy(Cookie::SameSite sameSite_) {
+    m_sameSite = sameSite_;
+}
+
+std::tm Cookie::expirationDate() const {
+    return m_expirationDate;
+}
+
+void Cookie::setExpirationDate(const std::tm& date) {
+    m_expirationDate = date;
+}
+
+std::string Cookie::domain() const {
+    return m_domain;
+}
+
+void Cookie::setDomain(const std::string& domain_) {
+    m_domain = domain_;
+}
+
+std::string Cookie::path() const {
+    return m_path;
+}
+
+void Cookie::setPath(const std::string& path_) {
+    m_path = path_;
+}
+
+std::string Cookie::name() const {
+    return m_name;
+}
+
+void Cookie::setName(const std::string& cookieName) {
+    m_name = cookieName;
+}
+
+std::string Cookie::value() const {
+    return m_value;
+}
+
+void Cookie::setValue(const std::string& value_) {
+    m_value = value_;
+}
+
+Cookie::SameSite Cookie::sameSiteFromString(std::string str) {
+    if (str == "none")
+        return Cookie::SameSite::None;
+    if (str == "lax")
+        return Cookie::SameSite::Lax;
+    if (str == "strict")
+        return Cookie::SameSite::Strict;
+    return Cookie::SameSite::Default;
+}
+
+std::string Cookie::sameSiteToString(SameSite samesite) {
+    switch (samesite) {
+    case Cookie::SameSite::None:
+        return "None";
+    case Cookie::SameSite::Lax:
+        return "Lax";
+    case Cookie::SameSite::Strict:
+        return "Strict";
+    case Cookie::SameSite::Default:
+        break;
+    }
+    return "";
+}
+
+/*
+ * see: https://searchfox.org/mozilla-central/source/nsprpub/pr/src/misc/prtime.c#931
+ * We only recognize the abbreviations of a small subset of time zones
+ * in North America, Europe, and Japan.
+ *
+ * PST/PDT: Pacific Standard/Daylight Time
+ * MST/MDT: Mountain Standard/Daylight Time
+ * CST/CDT: Central Standard/Daylight Time
+ * EST/EDT: Eastern Standard/Daylight Time
+ * AST: Atlantic Standard Time
+ * NST: Newfoundland Standard Time
+ * GMT: Greenwich Mean Time
+ * BST: British Summer Time
+ * MET: Middle Europe Time
+ * EET: Eastern Europe Time
+ * JST: Japan Standard Time
+ */
+typedef enum {
```

Comment:
```suggestion
enum TimeToken {
```

Is sufficient, there's no need for the typedef